src/plasma_manager.c

```diff
@@ -707,20 +704,33 @@ void process_message(event_loop *loop,
   free(req);
 }
 
-void new_client_connection(event_loop *loop,
-                           int listener_sock,
-                           void *context,
-                           int events) {
+client_connection *new_client_connection(event_loop *loop,
+                                         int listener_sock,
+                                         void *context,
+                                         int events) {
   int new_socket = accept_client(listener_sock);
   /* Create a new data connection context per client. */
   client_connection *conn = malloc(sizeof(client_connection));
   conn->manager_state = (plasma_manager_state *) context;
+  conn->cursor = 0;
   conn->transfer_queue = NULL;
   conn->fd = new_socket;
   conn->active_objects = NULL;
   conn->num_return_objects = 0;
   event_loop_add_file(loop, new_socket, EVENT_LOOP_READ, process_message, conn);
   LOG_DEBUG("New plasma manager connection with fd %d", new_socket);
+  return conn;
+}
+
+void _new_client_connection(event_loop *loop,
```

Comment:
This is ugly, but there doesn't seem a way around :(

A nicer way: Call this one "handle_new_client" and get rid of the void cast ;)